Sources/FusionBluetooth_Common/BluetoothManager.swift

```diff
@@ -0,0 +1,160 @@
+    
+    /**
+     *  @property uuid
+     *
+     *  @discussion The identifier in iOS and MAC address in Android.
+     *
+     */
+    public let uuid: String
+    
+     /**
+     *  @property isConnected
+     *
+     *  @discussion The state whether the peripheral is connected or not
+     *
+     */
+    public var isConnected: Bool
+    
+    public init(name: String?, uuid: String, isConnected: Bool) {
+        self.name = name
+        self.uuid = uuid
+        self.isConnected = isConnected
+    }
+}
+
+/**
+ *  @enum BMError
+ *
+ *  @discussion Error State
+ *
+ */
+public enum BMError: Equatable, Error {
+    case unknown // State unknown, update imminent.
+    case unsupported // The platform doesn't support the Bluetooth Low Energy Central/Client role.
+    case unauthorized // The application is not authorized to use the Bluetooth Low Energy role.
+    case discovering // The application is currently discovering the Bluetooth Device.
+    case resetting // The connection with the system service was momentarily lost, update imminent.
+    case notEnabled // Bluetooth is currently not enabled.
+    case notFound // The application could not find the Bluetooth Device.
+    case error(String)
+    
+    public func description() -> String {
+        switch self {
+        case .unknown:
+            return "Unexpected error"
+        case .unsupported:
+            return "The platform doesn't support the Bluetooth Low Energy Central/Client role."
+        case .unauthorized:
+            return "The application is not authorized to use the Bluetooth Low Energy role."
+        case .discovering:
+            return "The application is currently discovering the Bluetooth Device."
+        case .resetting:
+            return "The connection with the system service was momentarily lost, update imminent."
+        case .notEnabled:
+            return "Bluetooth is currently not enabled."
+        case .notFound:
+            return "The application could not find the Bluetooth Device."
+        case .error(let error):
+            return error
+        }
+    }
+}
+
+/**
+ *  @protocol BluetoothManagerProtocol
+ *
+ *  @discussion The delegate of managing Bluetooth.
+ *
+ */
+public protocol BluetoothManagerProtocol {
+    /*
+     * @method startDiscovering:
+     *
+     * @param receiver Returns a Peripheral discovered.
+     *
+     * @discussion Start Discovering Peripherals.
+     */
+    func startDiscovering(receiver: @escaping (Peripheral?, BMError?) -> Void)
+    
+    /*
+     * @method stopDiscovering:
+     *
+     * @discussion Stops scanning peripherals.
+     */
+    func stopDiscovering()
+        
+    /*
+     * @method connectDevice:
+     *
+     * @param peripheral Peripheral
+     * @param receiver Returns the success or error of Peripheral connection.
+     *
+     * @discussion Connects a peripheral
+     */
+    func connectDevice(peripheral: Peripheral, receiver: @escaping (Bool, BMError?) -> Void)
```

Comment:
Is there any logic behind not to rename Peripheral to Device and not to move the "*Device" functions to the the struct ? 

PLEASE, DO NOT TOUCH ANY IMPLEMENTATION UNTIL WE ARE NOT AGREED WITH THIS FILE !!!